sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
+            id: true,
+            startDate: true,
+          },
+        },
+        assessmentDate: true,
+      },
+      relations: {
+        application: {
+          student: { sinValidation: true, user: true },
+          programYear: true,
+        },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
+    if (!assessment) {
+      throw new Error(`Assessment if ${assessmentId} not found.`);
+    }
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    // Get the first assessment date for the current application. If there are multiple
+    // assessments for the current application, then set the first assessment date to
+    // the assessment date of the first assessment.
+    let firstAssessmentDate = getISODateOnlyString(assessment.assessmentDate);
+    const firstAssessment = await this.dataSource.manager
+      .getRepository(StudentAssessment)
+      .createQueryBuilder("assessment")
+      .select("assessment.assessmentDate")
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId: assessment.id })
+      .where("application.applicationNumber = :applicationNumber", {
+        applicationNumber: assessment.application.applicationNumber,
+      })
+      .orderBy("assessment.assessmentDate")
+      .getOne();
+    if (firstAssessment) {
+      firstAssessmentDate = getISODateOnlyString(
+        firstAssessment.assessmentDate,
+      );
+    }
+    const sfasApplicationLegacyRepo =
+      sfasApplicationType === "sfasPTApplication"
+        ? this.sfasPartTimeApplicationsRepo
+        : this.sfasApplicationsRepo;
+    const sfasApplicationLegacyAwards = sfasApplicationLegacyRepo
+      .createQueryBuilder(`${sfasApplicationType}`)
+      .select(`SUM(${sfasApplicationType}.csgp_award)`, "CSGP")
+      .addSelect(`SUM(${sfasApplicationType}.sbsd_award)`, "SBSD")
+      .addSelect(`SUM(${sfasApplicationType}.csgd_award)`, "CSGD")
+      .addSelect(`SUM(${sfasApplicationType}.bcag_award)`, "BCAG");
+    if (sfasApplicationType === "sfasPTApplication")
+      sfasApplicationLegacyAwards.addSelect(
+        `SUM(${sfasApplicationType}.cspt_award)`,
+        "CSPT",
+      );
+    sfasApplicationLegacyAwards
+      .innerJoin(`${sfasApplicationType}.individual`, "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where(`${sfasApplicationType}.startDate >= :startDate`, {
+            startDate: programYearStartDate,
+          }).andWhere(`${sfasApplicationType}.endDate <= :endDate`, {
```

Comment:
@andrewsignori-aot @guru-aot @andrepestana-aot @sh16011993 

Should this be `endDate<firstAssessmentDate` or  `endDate<=firstAssessmentDate`

Reply:
SIM considers applications in the past exclusive (`<`), hence it should be `>=` and `<`.
[Source code reference](https://github.com/bcgov/SIMS/blob/e8dc452c773c5a01c63c2072f98f668e328a6568/sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.models.ts#L84).

Reply:
From the other comment, got the endDate must not be used for logic.